gpytorch/settings.py

```diff
@@ -137,6 +137,22 @@ class detach_test_caches(_feature_flag):
     _state = True
 
 
+class deterministic_probes(_feature_flag):
+    """
+    Whether or not to resample probe vectors every iteration of training. If True, we use the same set of probe vectors
+    for computing log determinants each iteration. This introduces small amounts of bias in to the MLL, but allows us
+    to compute a deterministic estimate of it which makes optimizers like L-BFGS more viable choices.
+    """
```

Comment:
Let's add a warning here that this will cause weird behavior when training multiple models with this setting on b/c of the global scope of the caching.

Reply:
Added a warning comment. If someone finds the setting, they'll see.